src/server/middleware/auth.ts

```diff
@@ -523,12 +523,23 @@ const hashRequestBody = (req: FastifyRequest): string => {
  * @returns boolean
  * @async
  */
-const checkIpInAllowlist = async (req: FastifyRequest) => {
+const checkIpInAllowlist = async (
+  req: FastifyRequest,
+): Promise<
+  { isAllowed: false; ip: string } | { isAllowed: true; ip?: string }
+> => {
   const config = await getConfig();
-
   if (config.ipAllowlist.length === 0) {
-    return true;
+    return { isAllowed: true };
+  }
+
+  let ip = req.ip;
+  if (req.headers["cf-connecting-ip"]) {
```

Comment:
can we check for env.TRUST_PROXY or similar?
engine code is open source. An attacker can learn that we "blindly" trust this header and can circumvent IP allowlist for self-hosted engines

Reply:
Good catch